### PR TITLE
open-code vector copy since pcl changed sign of vertex number in v1.11

### PR DESCRIPTION
**Pull request by pierrewillenbrockdfki**

### CODE REVIEW

**Comment by pierrewillenbrockdfki** [comment]
@skasperski 

The relevant types are `std::vector<uint32_t> wrappers::PCLVertices::vertices` and `pcl::Indices pcl::Vertices::vertices` which is `using Indices = IndicesAllocator<>` which is `using IndicesAllocator = std::vector<index_t, Allocator>`, with 
```
  /**
   * \brief Type used for an index in PCL
   *
   * Default index_t = int for PCL 1.11, std::int32_t for PCL >= 1.12
   */
  using index_t = detail::int_type_t<detail::index_type_size, detail::index_type_signed>;
```

pcl 1.10 uses just `std::vector<std::uint32_t> pcl::Vertices::vertices`

**Comment by skasperski** [comment]
Looks good to me, but I cannot merge.

**Comment by pierrewillenbrockdfki** [comment]
@planthaber @chhtz

File: typekit/Opaques.cpp [hunk 1]
```diff
@@ -13,11 +13,19 @@
 
 void orogen_typekits::toIntermediate(::wrappers::PCLVertices& intermediate, ::pcl::Vertices const& real_type)
 {
-    intermediate.vertices = real_type.vertices;
+    size_t size = real_type.vertices.size();
+    intermediate.vertices.resize(size);
+    for(size_t i = 0; i < size; i++) {
+        intermediate.vertices[i] = real_type.vertices[i];
+    }
```
Comment (chhtz) [inline typekit/Opaques.cpp]: This works as well (and probably more likely to result in a memcpy:

    intermediate.vertices.assign(real_type.vertices.begin(), real_type.vertices.end());

File: typekit/Opaques.cpp [hunk 2]
```diff
@@ -13,11 +13,19 @@
...
 }
 
 void orogen_typekits::fromIntermediate(::pcl::Vertices& real_type, ::wrappers::PCLVertices const& intermediate)
 {
-    real_type.vertices = intermediate.vertices;
+    size_t size = intermediate.vertices.size();
+    real_type.vertices.resize(size);
+    for(size_t i = 0; i < size; i++) {
+        real_type.vertices[i] = intermediate.vertices[i];
+    }
```
Comment (chhtz) [inline typekit/Opaques.cpp]: Here as well:

    real_type.vertices.assign(intermediate.vertices.begin(), intermediate.vertices.end());

**Review by chhtz — COMMENTED** [review]
I guess keeping the `std::vector<uint32_t>` definition in wrappers/PCLVertices.hpp is still beneficial, to not break existing log-files?
Also, if I understand PCL correctly, it is possible to change the type of Vertices in newer versions (no idea, why they decided to change the default to signed).

File: typekit/Opaques.cpp (hunk 1 above)
Comment (pierrewillenbrockdfki) [reply typekit/Opaques.cpp]: I'll check if that still compiles, but to not get into repeated inserts(), one should still resize the vector beforehand.

**Comment by pierrewillenbrockdfki** [comment]
> I guess keeping the `std::vector<uint32_t>` definition in wrappers/PCLVertices.hpp is still beneficial, to not break existing log-files? Also, if I understand PCL correctly, it is possible to change the type of Vertices in newer versions (no idea, why they decided to change the default to signed).

I guess these are indices into a vertex array.

**Review by chhtz — APPROVED** [review]

**Comment by chhtz** [comment]
> > I guess keeping the `std::vector<uint32_t>` definition in wrappers/PCLVertices.hpp is still beneficial, to not break existing log-files? Also, if I understand PCL correctly, it is possible to change the type of Vertices in newer versions (no idea, why they decided to change the default to signed).
> 
> I guess these are indices into a vertex array.

Doesn't quite explain to me, why defaulting to signed indices is better (but that does not matter for this PR).

**Pull request merged**